### DISCUSSION TITLE
Signatures based on email domain

### DISCUSSION THREAD

**Discussion by dakolta** [Set-OutlookSignatures/Set-OutlookSignatures#55] [General]
How can you do a separate signature based on the domain of the primary email address.
I saw this section:
SmtpAddress, -:SmtpAddress
Make this template specific for the assigned e-mail address (all SMTP addresses of a mailbox are considered, not only the primary one)
The '-:' prefix makes this template invalid for the specified e-mail address.
Examples: office@example.com, -:test@example.com

But what we are needing is to have the signature only work for the primary email address, are we going to have to separate them as groups?

**Comment by GruberMarkus** [comment]
Set-OutlookSignatures supports filtering by group members and by e-mail address. In both cases, wildcards are not allowed.

To reach your goal, you have to create a group, fill it with members based on your criteria, and finally assign that group to a template.

If you are using the latest version of Set-OutlookSignatures, the group type does not really matter: It can be a security or distribution group, the mailbox user can be a direct or indirect member of the group.
In Exchange Online, the group can be a dynamic group (not supported on-prem).
On-prem, the group can be in a trusted domain too. When you are on-prem and have a Root-Child domain model, don't use the group type domain local until a new version fixing a bug around this group type is released.

**Comment by GruberMarkus** [comment]
Update: This is possible in the latest versions of Set-OutlookSignatures, as template assign can refer to a (custom) replacement variable.

Search the README for 'Make this template specific for the assigned replacement variable' to see details and an example.
